probe-rs/src/architecture/xtensa/communication_interface.rs

```diff
@@ -414,18 +418,27 @@ impl<'probe> XtensaCommunicationInterface<'probe> {
     }
 
     /// Steps the core by one instruction.
-    pub fn step(&mut self, by: u32, intlevel: u32) -> Result<(), XtensaError> {
+    pub fn step(&mut self, by: u32) -> Result<(), XtensaError> {
         // Instructions executed below icountlevel increment the ICOUNT register.
-        self.schedule_write_register(ICountLevel(intlevel + 1))?;
+        self.schedule_write_register(ICountLevel(self.core_properties.max_user_handler_level + 1))?;
```

Comment:
We count all user code - this means single stepping can step into an interrupt handler in theory, which prevents timing out on a long-running interrupt handler (and ending up somewhere in the middle of the handler).

Reply:
Reverted - this usually meant we stepped into a naked handler which is not useful at all. Better to try and skip over the interrupt handler.